client/add/add.ctrl.js

```diff
@@ -123,11 +127,18 @@ module.exports = class Add {
            toastAction,
            toastAction ? `/pokemon/${successfulUploads[0].created.id}` : undefined);
       })
-      .filter(line => line.success && line.created.box === box.data.id)
-      .then(lines => lines.slice(0, BOX_PAGE_SIZE - (box.data.contents.length % BOX_PAGE_SIZE)))
+      .filter(line => line.success)
       .map(line => line.created)
-      .then(lines => box.data.contents.push(...lines))
-      .tap(box.onscroll)
+      .each(line => {
+        if (this.boxesById[line.box].contents.length % BOX_PAGE_SIZE) {
+          this.boxesById[line.box].contents.push(line);
+        }
+        if (selectedBox && line.box === selectedBox.id
+          && selectedBox.data.contents.length % BOX_PAGE_SIZE) {
+          selectedBox.data.contents.push(line);
```

Comment:
It is not. The boxes are actually different boxes. Selectedbox is only when we have a #/box page open, and it is a different instance from the card boxes in this.boxes, which we always have (and those are what we see on the index page).

So, if we are on box 3, and we also have boxes 1 and 2, this.boxes will contain the 3 card box instances (1, 2 and 3), but selectedBox will not point to the same instance of 3 that is in this.boxes, because we create a new one when we navigate to that page.

Should we change how this works? Probably, but that's really not in the scope of this PR, as it's an existing issue. We also now wipe out selectedBox when we navigate to the homepage, so we only have a selectedBox when we are on a box page. (Also, I fixed a bug that did include duplicating pokemon because the box cards set themselves to be selectedBox, which is clearly nonsense).